### PR TITLE
Fix assertion for subject_type in creating pull request review comment

### PR DESCRIPTION
**Pull request by vanya20074**
Fix assertion for subject_type 
https://github.com/PyGithub/PyGithub/issues/2640
https://github.com/github/docs/issues/26844

### CODE REVIEW

**Comment by EnricoMi** [comment]
Can we also fix `side` and `start_side` while we are touching this?

**Comment by vanya20074** [comment]
> Can we also fix `side` and `start_side` while we are touching this?

I have tested lowercase value for `side` and `start_side` and got this error in response:

`right is not a member of [\"LEFT\", \"RIGHT\"]`

It can be concluded that in this case the values in the documentation are indicated correctly and must be in uppercase.

**Comment by EnricoMi** [comment]
In that case, we should relax the assertion to

    subject_type.lower() in ["line", "file", "side"]

so that both upper and lowercase is accepted by PyGithub and Github response tells users which is right.

I think they will have to harmonize those consts to either always lowercase or upper case them, and in order not to break user code, they will go for case-insensitive consts.

**Comment by vanya20074** [comment]
> In that case, we should relax the assertion to
> 
> ```
> subject_type.lower() in ["line", "file", "side"]
> ```
Done

**Comment by EnricoMi** [comment]
I meant relax the assertion for all string consts (`side`, `start_side`), so that our code looks consistent in some sense.

File: github/PullRequest.py [hunk 1]
```diff
@@ -447,9 +447,9 @@ def create_review_comment(
             "RIGHT",
         ], start_side
         assert in_reply_to is github.GithubObject.NotSet or isinstance(in_reply_to, int), in_reply_to
-        assert subject_type is github.GithubObject.NotSet or subject_type in [
-            "LINE",
-            "FILE",
+        assert subject_type is github.GithubObject.NotSet or subject_type.lower() in [
```
Comment (EnricoMi) [inline github/PullRequest.py:450]: On a second thought, lets remove this `lower` so we encode the current known API requirement:
```suggestion
        assert subject_type is github.GithubObject.NotSet or subject_type in [
```
Once docs are updated, we can fix all string assertions.

**Pull request merged**